src/LanguageServer/Protocol/Protocol/DocumentUri.cs

```diff
@@ -136,4 +136,17 @@ public override int GetHashCode()
             return this.ParsedUri.GetHashCode();
         }
     }
+
+    public static bool operator ==(DocumentUri? uri1, DocumentUri? uri2)
+    {
+        if (uri1 is null)
+            return uri2 is null;
+        else if (uri2 is null)
+            return false;
+
+        return uri1.Equals(uri2);
+    }
```

Comment:
```suggestion
    public static bool operator ==(DocumentUri? uri1, DocumentUri? uri2)
        => (uri1, uri2) switch
        {
            (null, null) => true,
            (null, _) or (_, null) => false,
            _ => uri1.Equals(uri2)
        };
```

Reply:
switched to using the pattern. I'm not too opinionated on this as the logic is pretty straightforward either way.